operator/pkg/ciliumendpointbatch/manager.go

```diff
@@ -0,0 +1,337 @@
+// Copyright 2021 Authors of Cilium
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciliumendpointbatch
+
+import (
+	"fmt"
+	"math/rand"
+	"time"
+
+	meta_v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	"github.com/cilium/cilium/pkg/annotation"
+	cilium_v2 "github.com/cilium/cilium/pkg/k8s/apis/cilium.io/v2alpha1"
+	"github.com/cilium/cilium/pkg/lock"
+)
+
+var (
+	// sequentialLetters contains lower case alphabets without vowels and few numbers.
+	// skipped vowels and numbers [0, 1] to avoid generating controversial names.
+	sequentialLetters = []rune("bcdfghjklmnpqrstvwxyz2456789")
+)
+
+// ceBatch holds local copy of CiliumEndpointBatch.
+type ceBatch struct {
+	ceb          *cilium_v2.CiliumEndpointBatch
+	backendMutex lock.RWMutex
+}
+
+// cebManager
+type cebManager interface {
+	InsertCepInCache(cep *cilium_v2.CoreCiliumEndpoint) (string, error)
+	RemoveCepFromCache(cep *cilium_v2.CoreCiliumEndpoint) error
+	getCebFromCache(cebName string) (*cilium_v2.CiliumEndpointBatch, error)
+	findCeb(cep *cilium_v2.CoreCiliumEndpoint) (*ceBatch, bool)
+	updateCebInCache(ceb *cilium_v2.CiliumEndpointBatch, deepCopy bool)
+	deleteCebFromCache(cebName string)
+	createCeb(name string) *ceBatch
+	getCepCount() int
+	getCebCount() int
+	getAllCeps() map[string]*cilium_v2.CoreCiliumEndpoint
+}
```

Comment:
My impression is that the callers of this interface don't use all methods but simply:
```golang
type cebManager interface {
	InsertCepInCache(cep *cilium_v2.CoreCiliumEndpoint) (string, error)
	RemoveCepFromCache(cep *cilium_v2.CoreCiliumEndpoint) error
	getCebFromCache(cebName string) (*cilium_v2.CiliumEndpointBatch, error)
	updateCebInCache(ceb *cilium_v2.CiliumEndpointBatch, deepCopy bool)
	deleteCebFromCache(cebName string)
	createCeb(name string) *ceBatch
	getCebCount() int
	getAllCeps() map[string]*cilium_v2.CoreCiliumEndpoint
}
```

What's the rational to have the interface in the same place where the interface's implementation exists?

Reply:
The reasoning behind having interface here is, in future we may have more than one algorithm to patch CEP's into CEB. Eventually end user use only one method/algorithm to pack CEPs into CEB. `cebManager` Interface will remain common to all algorithms and each algorithm can have it own data-structure to pack CEPs in CEB.